pkg/cmd/pkg/package_list.go

```diff
@@ -19,149 +21,175 @@ import (
 	"github.com/spf13/cobra"
 )
 
+type listOptions struct {
+	appID string
+	query string
+
+	cliOnly    bool
+	jsonOutput bool
+}
+
 func newCmdPackageList(ctx api.Context) *cobra.Command {
-	var cliOnly, jsonOutput bool
-	var appID string
+	var opts listOptions
 
 	cmd := &cobra.Command{
 		Use:   "list",
 		Short: "Print a list of the installed DC/OS packages",
-		Args:  cobra.RangeArgs(0, 1),
+		Args:  cobra.MaximumNArgs(1),
 		PreRunE: func(cmd *cobra.Command, args []string) error {
-			if appID != "" {
+			if opts.appID != "" {
 				if len(args) > 0 {
 					return errors.New("cannot use the flags `--app-id` and an argument at the same time")
 				}
 
-				if cliOnly && !jsonOutput {
+				if opts.cliOnly && !opts.jsonOutput {
 					return errors.New("cannot use the flags `--app-id` and `--cli` at the same time")
 				}
 			}
 			return nil
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			var list []cosmos.Package
-			pkgNames := make(map[string]bool)
-
-			// Get the commands installed locally, those do not have app IDs thus we can skip if one was provided using `app-id`.
-			cluster, err := ctx.Cluster()
+			if len(args) == 1 {
+				opts.query = args[0]
+			}
+			c, err := cosmos.NewClient(ctx, pluginutil.HTTPClient(""))
 			if err != nil {
 				return err
 			}
+			return listPackages(ctx, opts, c)
+		},
+	}
+	cmd.Flags().StringVar(&opts.appID, "app-id", "", "The application ID")
+	cmd.Flags().BoolVar(&opts.cliOnly, "cli", false, "Command line only")
```

Comment:
Hmm I think the description here should be a little clearer. What does "Command line only" mean? Does it disable some web interface or something?